packages/badge/components/badgeButton.tsx

```diff
@@ -10,7 +11,7 @@ export interface IBadgeButtonProps {
     | "warning"
     | "danger"
     | "outline";
-  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+  onClick: (event?: React.SyntheticEvent<HTMLElement>) => void;
```

Comment:
so `onClick` is required, but its parameter isnt? 🤔 shouldnt both be required? I mean, its `Clickable` … something should happen, after you clicked…

Reply:
Something would happen... it just means that you might not get an event object.

Reply:
It just means that the parameter is optional, so if you pass in a function without a parameter defined. Let's pretend we do have a kill service button. You click it and the application will know by context what it should do, there is no need to look at the event, right? But in some other cases, it might be necessary to look at the event. That is why it is kept optional.